keps/sig-windows/2258-kubectl-service-log-viewer/README.md

```diff
@@ -0,0 +1,213 @@
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+A Kubernetes cluster administrator has to log in to the relavant control-plane
+or worker nodes to view the logs of the API server, kubelet etc. Or they would
+have to implement a client side reader. A simpler and more elegant method would
+be to allow them to use the kubectl CLI to also view these logs similar to
+using it for other interactions with the cluster.
+
+## Motivation
+
+Troubleshooting issues with control-plane and worker nodes typically requires 
+a cluster administrator to SSH into the nodes for debugging. While certain
+issues will require being on the node, issues with the kube-proxy or kubelet,
+to name a couple, could be solved by perusing their logs. However this
+too requires the administrator to SSH access into the nodes. Giving a way for
+them to view the logs using kubectl will significantly simplify their
+troubleshooting.
+
+
+### Goals
+Provide a cluster administrator with a streaming view of logs using kubectl
+without them having to implement a client side reader or logging into the node.
+This would work for:
+- Linux worker and control plane nodes using a distribution that have systemd
+  / journald support
+- Windows worker nodes (all supported variants)
+
+### Non-Goals
+Providing support for non-systemd Linux distributions.
```

Comment:
I guess this is obvious but maybe you can point out here that reporting logs for nodes/kubelets that have config/connection issues with the cluster is out-of-scope.

Reply:
Done